plenum/server/monitor.py

```diff
@@ -380,14 +395,7 @@ def requestOrdered(self, reqIdrs: List[Tuple[str, int]], instId: int,
 
             if key in requests:
                 identifier = requests[key].request.identifier
-                if identifier not in self.clientAvgReqLatencies[instId]:
-                    self.clientAvgReqLatencies[instId][identifier] = (0, 0.0)
-                totalReqs, avgTime = self.clientAvgReqLatencies[instId][identifier]
-                # If avg of `n` items is `a`, thus sum of `n` items is `x` where
-                # `x=n*a` then avg of `n+1` items where `y` is the new item is
-                # `((n*a)+y)/n+1`
-                self.clientAvgReqLatencies[instId][identifier] = (
-                    totalReqs + 1, (totalReqs * avgTime + duration) / (totalReqs + 1))
+                self.clientAvgReqLatencies[instId].add_duration(identifier, duration)
```

Comment:
Can it happen so that key is NOT in requests? If it can, but is not expected a lot - we can really add request here as well and merge `add_request` and `add_duration` as I already mentioned.